tests/snapshot.rs

```diff
@@ -41,7 +42,7 @@ fn test_build_graphs_and_create_embeddings() {
         // calculate embeddings in memory
         calculate_embeddings(
             config.clone(),
-            sparse_matrix,
+            sparse_matrix.clone(),
```

Comment:
maybe Arc::clone(&sparse_matrix) to indicate that this is cheap (though, maybe not)